### ISSUE TITLE
Test run recording problem

### ISSUE THREAD

**Issue by jaques-sam** [bondagit/aes67-linux-daemon#134]
It happened to me once, that I was able to record from the RAVENNA ALSA sink generated from the AES67 deamon, but not anymore. My sink seems to be configured correctly and same as before.

The `arecord` seems to be running forever. This doesn't look like a AES67 daemon issue, but I'm new to ALSA, so I need some advice here.

It seems in `run_tests.sh`, the same could happen without noticing the issue. It's hidden by the way the script is written.
It says:
```
arecord -M -d $SEC -D plughw:RAVENNA -f $SAMPLE_FORMAT -r $SAMPLE_RATE -c $CHANNELS -t raw /tmp/sink_test.raw   > /dev/null 2>&1 &
sleep 10
...
while killall -0 arecord ; do
  sleep 1
done
```

This means whatever duration SEC has, it's automatically killed after 10 seconds, which is obviously a mistake.
The error that `arecord` will dump, will not be visible because of `/dev/null`. 
For me `killall -0 arecord` doesn't even work. When I kill it I got `arecord: pcm_read:2145: read error: Interrupted system call` and the created file is empty, meaning this check `if [ -f /tmp/sink_test.raw ] ; then` is also not correct...

**Comment by bondagit** [comment]
I invite you to study the script a bit more and try to understand the behaviour before opening issues like this one.
`kill -0` (signal 0) is used to see if the process is still alive and not to terminate it.
SEC is the test duration in second computed from the argument number 4 of the script.

**Comment by jaques-sam** [comment]
Okay, sry `killall -h` didn't give me info about `-0` (I know -9, but also no info in the help, it's very cryptic :), but you're definitely right that part is correct. In any case, I didn't want it to label this topic as an "issue".
What I forgot to mention is that the tests run forever still, on `Playing raw data './test/test.raw'`. If you can give me some more info/doc about why arecord/aplay is running forever that would be helpful. ;-) Thanks!

**Comment by jaques-sam** [comment]
I tested on Debian 10 & Fedora 28.
Another day later, I was able to run the tests with success on Fedora, but I cannot say what has changed. The test scripts cleans up all the necessities so I don't know what caused it :( Actually did a reboot per day, which could've influenced the behavior.
The output of my first `run_test.sh` on Debian 10 (which ran already for 1 hour) is:
```
...
Starting to playback test on source ...
Playing raw data './test/test.raw' : Signed 16 bit Little Endian, Rate 48000 Hz, Stereo
```
Second time I ran it, the last output is the `insmod`, meaning the cleanup is taken forever and this is because `killall -q` doesn't quit the previous daemon:
```
confero@confero-gateway:~/repos/aes67-linux-daemon$ sudo killall -q aes67-daemon
confero@confero-gateway:~/repos/aes67-linux-daemon$ ps aux | grep aes67-daemon
root     12450  0.0  0.0 315996  7808 pts/0    Sl   17:50   0:01 ./daemon/aes67-daemon -c ./test/daemon_.conf
```
I could kill it with `kill -9 aes67-daemon` and tried again...
The test ran for another hour, without continuing after `Playing raw data...`.

[off-topic, just need advice] I have another issue on trying to play from a source (which is not done in your tests) using vlc (by loading the sdp), gstreamer (with udpsrc/rtpL16depay/autoaudiosink) or even from within Wireshark (Telephony > RTP player) which seems like valid RTP packets (repeating PT= L16/98 of length 246 & 118) but I'm unable to play them. Don't know if you have some doc or info about this... Thanks anyway!

Side note info: now I know why aplay/arecord could run forever (even the duration is a few seconds). It implies they don't receive packets (= standing still in time, which feels illogical to me).
